src/radical/pilot/controller/pilot_launcher_worker.py

```diff
@@ -718,8 +726,45 @@ def run(self):
                         jd.wall_time_limit       = runtime
                         jd.total_physical_memory = memory
                         jd.queue                 = queue
+                        jd.candidate_hosts       = candidate_hosts
                         jd.environment           = dict()
 
+                        # TODO: not all files might be required, this also needs to be made conditional
+                        if not shared_filesystem:
+                            jd.file_transfer = [
+                                #'%s > %s' % (bootstrapper_path, os.path.basename(bootstrapper_path)),
+                                '%s > %s' % (bootstrapper_path, os.path.join(jd.working_directory, 'input', os.path.basename(bootstrapper_path))),
+                                #'%s < %s' % ('agent.log', os.path.join(jd.working_directory, 'agent.log')),
+                                #'%s < %s' % (os.path.join(jd.working_directory, 'agent.log'), 'agent.log'),
+                                #'%s < %s' % ('agent.log', 'agent.log'),
+                                #'%s < %s' % (os.path.join(jd.working_directory, 'STDOUT'), 'unit.000000/STDOUT'),
+                                #'%s < %s' % (os.path.join(jd.working_directory, 'unit.000000/STDERR'), 'STDERR')
+                                #'%s < %s' % ('unit.000000/STDERR', 'unit.000000/STDERR')
+
+                                # TODO: This needs to go into a per pilot directory on the submit node
+                                '%s < %s' % ('pilot.0000.log.tgz', 'pilot.0000.log.tgz')
+                            ]
+
+                            if stage_sdist:
+                                jd.file_transfer.extend([
+                                    #'%s > %s' % (rp_sdist_path, os.path.basename(rp_sdist_path)),
+                                    '%s > %s' % (rp_sdist_path, os.path.join(jd.working_directory, 'input', os.path.basename(rp_sdist_path))),
+                                    #'%s > %s' % (saga.sdist_path, os.path.basename(saga.sdist_path)),
+                                    '%s > %s' % (saga.sdist_path, os.path.join(jd.working_directory, 'input', os.path.basename(saga.sdist_path))),
+                                    #'%s > %s' % (ru.sdist_path, os.path.basename(ru.sdist_path)),
+                                    '%s > %s' % (ru.sdist_path, os.path.join(jd.working_directory, 'input', os.path.basename(ru.sdist_path)))
+                                ])
+
+                            if agent_config:
+                                jd.file_transfer.append('%s > %s' % (cfg_tmp_file, os.path.join(jd.working_directory, 'input', agent_cfg_name)))
```

Comment:
We always have a cfg file.  I think your assumption is that in the default agent_config we won't need it on the other hosts?  Will that always be true?  Why do we then copy the dists?  I am probably missing the point here wrt to unavailable shared FS...